### ISSUE TITLE
Return all selected filters in dashboard

### ISSUE THREAD

**Issue by NyakudyaA** [kartoza/healthyrivers#249]
# Problem

All the filters selected, should be returned in the detailed dashboard and CSV file.

**Comment by NyakudyaA** [comment]
Dimas. Download should respect whatever filters are applied.

**Comment by dimasciput** [comment]
@NyakudyaA we already did that

**Comment by Vries14** [comment]
When I applied the reference category filter on "consulting report" and clicked on 'Anguilla mossambica" taxa, What I found was that downloading the csv from the detailed dashboard had mostly Published article in the reference category, and only 1 consulting report, but the filter was applied on consulting report(see screenshot below)
![csv_download](https://user-images.githubusercontent.com/42169058/47001659-4a96dd00-d12b-11e8-9db9-f6b238030922.png)


What is happening is that when a filter gets applied and clicking on taxa it gives me all the taxa records available, and not filtering by the specific search. eg: when apply a filter on a collector it does not filter the taxa based on the collector but showing all different types of taxa that the collector has uploaded. See from my screenshot that when clicking on  "Anguilla mossambica" , it gives me all 39 records instead of only the "Anguilla mossambica records that were added by "Abrahams. P" 
![filters](https://user-images.githubusercontent.com/42169058/47001743-7c0fa880-d12b-11e8-9927-3713d6ed6ece.png)


I think @NyakudyaA was talking about this?
